src/Network/Transport/TCP.hs

```diff
@@ -500,6 +501,10 @@ data TCPParameters = TCPParameters {
   , transportConnectTimeout :: Maybe Int
     -- | Create a QDisc for an EndPoint.
   , tcpNewQDisc :: forall t . IO (QDisc t)
+    -- | Optional maximum length (in bytes) for a peer's address.
+  , tcpMaxAddressLength :: Maybe Word32
+    -- | Optional maximum length (in bytes) to receive from a peer.
```

Comment:
Might as well add to the API docs here what the behaviour is, ie what you said in the PR description.

> If a peer attempts to send an address of length exceeding the limit, the connection will be refused (socket will close).

> If a peer attempts to send data on a lightweight connection exceeding the limit, the heavyweight connection which carries that lightweight connection will go down. The peer and the local node will get an EventConnectionLost.